### PR TITLE
Improve the customization of WebView on Android

### LINKED ISSUE #11004: Android WebViewClient's ShouldInterceptRequest is never called in MAUI WebView
### Description

Android `WebViewClient`'s `ShouldInterceptRequest` is never called in MAUI `WebView`

### Steps to Reproduce

1.create a maui demo;
2.copy the following code to `MauiProgram.cs`:

```cs
public static class MauiProgram
{
	public static MauiApp CreateMauiApp()
	{
		var builder = MauiApp.CreateBuilder();
		builder
			.UseMauiApp<App>()
			.ConfigureFonts(fonts =>
			{
				fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
				fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
			})
			.ConfigureMauiHandlers(handlers =>
			{
				handlers.AddHandler<Microsoft.Maui.Controls.WebView, ProblemHandler2>();
			});

		return builder.Build();
	}
}

internal class ProblemHandler1 : WebViewHandler
{

	protected override Android.Webkit.WebView CreatePlatformView()
	{
		var wv = base.CreatePlatformView();
		wv.SetWebViewClient(new CustomWebClient());

		return wv;
	}

}

internal class ProblemHandler2 : WebViewHandler
{

    protected override Android.Webkit.WebView CreatePlatformView()
    {
        var wv = new Android.Webkit.WebView(Android.App.Application.Context);
        wv.SetWebViewClient(new CustomWebClient());

        return wv;
    }

}

internal class CustomWebClient : WebViewClient
{

    public override WebResourceResponse ShouldInterceptRequest(Android.Webkit.WebView view, IWebResourceRequest request)
    {
        Debugger.Break();
        Debug.WriteLine(request.Url.ToString());

        return base.ShouldInterceptRequest(view, request);
    }

}
```

MainPage.xaml.cs

```cs
public partial class MainPage : ContentPage
{
	int count = 0;

	public MainPage()
	{
		InitializeComponent();

        WebViewHandler.Mapper.AppendToMapping("MyHandler", (handler, view) =>
        {
#if ANDROID
			var xWv = handler.PlatformView;

			// For ProblemHandler2, this is needed to actually navigate:
			xWv.LoadUrl("https://www.google.com/");
#endif
        });

        this.wv.Source = "https://www.google.com/";
	}

}
```

3.build and deploy this demo to android device. (I only tested on android 13 emulator )

### Link to public reproduction project repository

https://github.com/datvm/DemoMauiWvClient

### Version with bug

Unknown/Other (please specify)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android 13

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change

Generally, the WebView is customizable in the handlers, however the WebViewClient and WebChromeClient are less so. You _can_ override the values and set your own with the two mappers, but it does not work as expected as the handler assumes that it will control the clients.

This PR removes the fields for the clients and sets the handler field in the client to be a weak reference. This will allow the handler to be collected as well as the client when the handler and the webview are collected.

### Issues Fixed

Fixes #11004 

### Notes

The correct way to override the `WebViewClient` is to create a type that derives from `MauiWebViewClient` as that has some core navigation features:

```cs
class CustomWebClient : MauiWebViewClient
{
	public CustomWebClient(WebViewHandler handler)
		: base(handler)
	{
	}

	public override WebResourceResponse ShouldInterceptRequest(AWebView view, IWebResourceRequest request)
	{
		// TODO: some cool code

		return base.ShouldInterceptRequest(view, request);
	}
}
```

And then use that in the mappers:

```cs
builder
	.ConfigureMauiHandlers(_ =>
	{
		WebViewHandler.Mapper.ModifyMapping("WebViewClient", (handler, view, previous) =>
		{
			if (handler is WebViewHandler mauiHandler)
				handler.PlatformView.SetWebViewClient(new CustomWebClient(mauiHandler));
		});
	});
```

### CODE REVIEW

**Review by jsuarezruiz — APPROVED** [review]

File: src/Core/src/Handlers/WebView/WebViewHandler.Android.cs [hunk 1]
```diff
@@ -52,17 +49,20 @@ public override void SetVirtualView(IView view)
 
 		protected override void DisconnectHandler(AWebView platformView)
 		{
-			if (_webViewClient is MauiWebViewClient mauiWebViewClient)
-				mauiWebViewClient.Disconnect();
+			if (OperatingSystem.IsAndroidVersionAtLeast(26))
```
Comment (Redth) [inline src/Core/src/Handlers/WebView/WebViewHandler.Android.cs:52]: @mattleibow why are we now putting this behind an API 26 check?

**Review by Redth — APPROVED** [review]

**Comment by samhouts** [comment]
@PureWeen Can we backport this one manually?

**Comment by mattleibow** [comment]
Backport PR:

 - https://github.com/dotnet/maui/pull/16692

**Pull request merged**